test/e2e/plain-v0_provisioner_test.go

```diff
@@ -345,6 +345,139 @@ var _ = Describe("plain-v0 provisioner bundleinstance", func() {
 		})
 	})
 
+	// TODO: try to create a BundleInstance that references a Bundle that contains
+	// existing resources.
+	When("a BundleInstance depends on another BundleInstance", func() {
+		var (
+			bundle *rukpakv1alpha1.Bundle
+			bi     *rukpakv1alpha1.BundleInstance
+			ctx    context.Context
+		)
+		BeforeEach(func() {
+			ctx = context.Background()
+
+			By("creating the testing OLM APIs Bundle resource")
+			bundle = &rukpakv1alpha1.Bundle{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "olm-apis",
+				},
+				Spec: rukpakv1alpha1.BundleSpec{
+					ProvisionerClassName: plainProvisionerID,
+					Image:                "quay.io/tflannag/olm-plain-bundle:olm-api-v0.20.0",
+				},
+			}
+			err := c.Create(ctx, bundle)
+			Expect(err).To(BeNil())
+
+			bi = &rukpakv1alpha1.BundleInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "olm-apis",
+				},
+				Spec: rukpakv1alpha1.BundleInstanceSpec{
+					ProvisionerClassName: plainProvisionerID,
+					BundleName:           bundle.GetName(),
+				},
+			}
+			err = c.Create(ctx, bi)
+			Expect(err).To(BeNil())
+		})
+		AfterEach(func() {
+			By("deleting the testing Bundle resource")
+			Eventually(func() error {
+				return client.IgnoreNotFound(c.Delete(ctx, bundle))
+			}).Should(Succeed())
+
+			By("deleting the testing BundleInstance resource")
+			Eventually(func() error {
+				return client.IgnoreNotFound(c.Delete(ctx, bi))
+			}).Should(Succeed())
+		})
+
+		It("should project a failed installation state", func() {
+			By("projecting a failed installation state")
+			Eventually(func() bool {
```

Comment:
Is it possible to return `(bool, error)` from these things so that unexpected errors can short circuit the wait?